src/roster/views.py

```diff
@@ -20,28 +22,89 @@ def guest_login(request):
     login(request, clusive_user.user)
     return redirect('reader_index')
 
-
-def set_preference(request, pref, value):
-    user = ClusiveUser.from_request(request)
-    preference = user.get_preference(pref)
-    # Don't re-save again if the set preference value is the same as the request value
-    if(preference.value != value):
-        preference.value = value
-        preference.save()
-        preference_changed.send(sender=ClusiveUser.__class__, request=request, preference=preference)
-    return JsonResponse({'success' : 1})
-
-
-def get_preferences(request):
-    user = ClusiveUser.from_request(request)
-    prefs = user.get_preferences()
-    return JsonResponse({p.pref:p.value for p in prefs})
-
-# TODO: how does a preference reset get logged?
-def reset_preferences(request):
-    user = ClusiveUser.from_request(request)    
-    user.delete_preferences()
-    return JsonResponse({'success': 1})
+class PreferenceView(View):
+    def get(self, request):
+        user = ClusiveUser.from_request(request)
+        prefs = user.get_preferences()
+        prefs_processed = convert_preference_values(prefs)
+        return JsonResponse(prefs_processed)
+    def post(self, request):        
+        try:
+            request_prefs = json.loads(request.body)
+        except json.JSONDecodeError:
+            return JsonResponse({'success': 0, 'message': 'Invalid JSON in request'})
+
+        user = ClusiveUser.from_request(request)        
+        set_user_preferences(user, request_prefs, request)
+        for req_pref_key in request_prefs:
+            req_pref_val = request_prefs[req_pref_key]
+            preference = user.get_preference(req_pref_key)               
+            if(preference.value != req_pref_val):                           
+                preference.value = req_pref_val
+                preference.save()
+                preference_changed.send(sender=ClusiveUser.__class__, request=request, preference=preference)            
```

Comment:
Is this redundant?  Looks like it's the same as what set_user_preferences does.

Reply:
Yes, I cleverly factored that out into a function, then forgot to actually remove it. Fixed.